dotnet/Services/CybersourcePaymentService.cs

```diff
@@ -398,6 +430,96 @@ public async Task<CreatePaymentResponse> CreatePayment(CreatePaymentRequest crea
+                            LockTtl = selectedSla.LockTtl,
+                            Name = selectedSla.Name,
+                            PickupDistance = selectedSla.PickupDistance,
+                            PickupPointId = selectedSla.PickupPointId,
+                            PickupStoreInfo = new PickupStoreInfo
+                            {
+                                AdditionalInfo = selectedSla.PickupStoreInfo.AdditionalInfo,
+                                Address = selectedSla.PickupStoreInfo.Address != null ? new VtexAddress
+                                {
+                                    AddressId = selectedSla.PickupStoreInfo.Address.AddressId,
+                                    AddressType = selectedSla.PickupStoreInfo.Address.AddressType,
+                                    City = selectedSla.PickupStoreInfo.Address.City,
+                                    Complement = selectedSla.PickupStoreInfo.Address.Complement,
+                                    Country = selectedSla.PickupStoreInfo.Address.Country,
+                                    GeoCoordinates = selectedSla.PickupStoreInfo.Address.GeoCoordinates,
+                                    IsDisposable = selectedSla.PickupStoreInfo.Address.IsDisposable,
+                                    Neighborhood = selectedSla.PickupStoreInfo.Address.Neighborhood,
+                                    Number = selectedSla.PickupStoreInfo.Address.Number,
+                                    PostalCode = selectedSla.PickupStoreInfo.Address.PostalCode,
+                                    ReceiverName = selectedSla.PickupStoreInfo.Address.ReceiverName,
+                                    Reference = selectedSla.PickupStoreInfo.Address.Reference,
+                                    State = selectedSla.PickupStoreInfo.Address.State,
+                                    Street = selectedSla.PickupStoreInfo.Address.Street
+                                } : null,
+                                DockId = selectedSla.PickupStoreInfo.DockId,
+                                FriendlyName = selectedSla.PickupStoreInfo.FriendlyName,
+                                IsPickupStore = selectedSla.PickupStoreInfo.IsPickupStore
+                            },
+                            PolygonName = selectedSla.PolygonName,
+                            ShippingEstimate = selectedSla.ShippingEstimate,
+                            ShippingEstimateDate = selectedSla.ShippingEstimateDate,
+                            Price = selectedSla.Price,
+                            Tax = selectedSla.Tax,
+                            TransitTime = selectedSla.TransitTime
+                        };
+
+                        requestWrapper.Shipping.DeliveryIds.AddRange(selectedSla.DeliveryIds);
+                        requestWrapper.Shipping.CourierName = string.Join(", ", selectedSla.DeliveryIds.Select(d => d.CourierName).Distinct().ToArray());
```

Comment:
since we are converting the whole thing to a string anyways, we can skip .toArray and just do a forEach and do string concatenation on each distinct element.

Reply:
Join is more efficient than string concatenation.  Changing a collection to an array should not be too expensive.